powerline/renderers/lemonbar.py

```diff
@@ -36,10 +37,26 @@ def hl(self, contents, fg=None, bg=None, attrs=None):
 		return text + contents + '%{F-B--u}'
 
 	def render(self, *args, **kwargs):
+		self.segment_info['output'] = kwargs.get('matcher_info')
```

Comment:
Missed that modifying `self.segment_info` is not thread-safe. With default implementation of `Renderer.get_segment_info()` method since you do not supply `segment_info` in kwargs this line should be missing and next lines should contain `segment_info={'output': kwargs.get('matcher_info')}`.